oat-sa/extension-tao-itemqti
# Feature/AUT-3790/Resource manager deletion hook

### Related to: https://oat-sa.atlassian.net/browse/AUT-3790

Requires: 
 - [x] https://github.com/oat-sa/tao-core-ui-fe/pull/601
 - [x] https://github.com/oat-sa/tao-core/pull/4092

### Summary

Allow to hook the file deletion in the resource manager.
When deleting a CSS file that is attached to the current item, as the user first and save the item upon file deletion.

### Details

The file deletion in the resource manager was relying on the `ui/deleter` helper, which is a global helper watching for elements having the `data-delete` attribute and adding behavior to them. However, it was impossible to hook it and insert a validation step in the middle. While this is possible to inject a step between the deletion event and the file deletion, the deletion event is emitted after the line is removed from the DOM. For this is reason, it is better to break the dependency to the `ui/deleter` helper and write a specific handling directly to the fileSelector component, adding an option to hook it.

### How to test

- make sure to also install the companion PR https://github.com/oat-sa/tao-core/pull/4092
- checkout the branch: `git checkout -t origin/feature/AUT-3790/resource-manager-deletion-hook`
- create an item or use an existing one
- open the item for authoring
- open the style editor (top right corner of the editor)
- click the add stylesheet button
- upload a stylesheet and attach it to the item
- click the add stylesheet button
- delete the stylesheet: a confirm dialog should be displayed
  - if you click cancel, nothing will happen
  - if you click ok, the file is deleted, removed from the item, and the item is saved

<video src="https://github.com/user-attachments/assets/0337d643-71dc-4f53-a810-ff18c738a91e"></video>
